sql/src/main/java/io/crate/planner/consumer/UpdateConsumer.java

```diff
@@ -99,11 +100,18 @@ public PlannedAnalyzedRelation visitDocTableRelation(DocTableRelation relation,
                 if (whereClause.docKeys().isPresent()) {
                     if (upsertById == null) {
                         Tuple<String[], Symbol[]> assignments = Assignments.convert(nestedAnalysis.assignments());
+                        int numBulkResponses = statement.nestedStatements().size();
+                        if (numBulkResponses == 1) {
+                            // disable bulk logic for 1 bulk item
+                            numBulkResponses = 0;
+                        }
                         upsertById = new UpsertById(
                             plannerContext.jobId(),
                             plannerContext.nextExecutionPhaseId(),
                             false,
-                            statement.nestedStatements().size() > 1, assignments.v1(),
+                            numBulkResponses,
+                            new ObjectIntHashMap<String>(),
```

Comment:
Is there a case where this map isn't just an empty new instance? If not, this could be done within the constructor